README.md

```diff
@@ -443,32 +443,72 @@ of the time a higher-level object will suffice as an argument.
 **Bad:**
 
 ```php
-function createMenu(string $title, string $body, string $buttonText, bool $cancellable): void
-{
-    // ...
+class Questionnaire
+{
+    public function __construct(
+        string $firstname,
+        string $lastname,
+        string $patronymic,
+        string $region,
+        string $district,
+        string $city,
+        string $phone,
+        string $email
+    ) {
+        // ...
+    }
 }
 ```
 
 **Good:**
 
 ```php
-class MenuConfig
+class Name
 {
-    public $title;
-    public $body;
-    public $buttonText;
-    public $cancellable = false;
+    private $firstname = '';
```

Comment:
There is a difference between this

```php
class Name
{
    /**
     * @var string
     */
    private $firstname = '';

    public function __construct(string $firstname)
    {
        $this->firstname = $firstname;
    }
}
```

and this

```php
class Name
{
    /**
     * @var string|null
     */
    private $firstname;

    public function __construct(string $firstname)
    {
        $this->firstname = $firstname;
    }
}
```
Especially given the [changes](https://wiki.php.net/rfc/typed_properties_v2) that we expect in PHP 7.4.

Reply:
I don't agree with that.
In PHP 7.4 you won't be able to retrieve properties that have not been set.
```
<?php

class A {
    public string $b;
}
$a = new A();
//$a->b = '';
var_dump($a->b);
```

Will give you `Fatal error: Uncaught Error: Typed property A::$b must not be accessed before initialization`, but if you remove the comment it will work as expected.

This because you MUST specify that it can be nullable if you want it.

This one is a private property that can only be set from inside, all the getters will throw an error in the same way as PHP7.4 will if you are not setting the variable :D

Reply:
> If a typed property does not have a default value, no implicit null default value is implied (even if the property is nullable). Instead, the property is considered to be uninitialized. Reads from uninitialized properties will generate a TypeError (unless __get() is defined, see next section).

Hm... I didn't read the RFС very carefully. This is unexpected. This noticeably changes the behavior.

```php
class A
{
    /**
     * @var string|null
     */
    public $a;

    public ?string $b;
}
$a = new A();
var_dump($a->a); // NULL
var_dump($a->b); // Fatal error: Uncaught Error: Typed property A::$b must not be accessed before initialization
```

Reply:
I prefer to explicitly specify the data type, but due to changes in PHP 7.4, this may no longer be relevant.